### PR TITLE
filter expired entities when reading from handles

### PR DESCRIPTION
**Pull request by galganif**

### CODE REVIEW

**Comment by galganif** [comment]
cl/305618875

File: java/arcs/core/entity/StorageAdapter.kt [hunk 1]
```diff
@@ -62,4 +70,7 @@ class ReferenceStorageAdapter<E : Entity>(
     override fun referencableToStorable(referencable: StorageReference): Reference<E> {
         return Reference(entitySpec, referencable)
     }
+
+    // Reference handle expiration is not supported.
+    override fun isExpired(value: Reference<E>): Boolean = false
```
Comment (mmandlis) [inline java/arcs/core/entity/StorageAdapter.kt:75]: add a TODO to support `isExpired` for References? A Reference may expire before the underlying Entity does.

File: java/arcs/core/entity/EntityBase.kt [hunk 2]
```diff
@@ -41,7 +41,7 @@ open class EntityBase(
         private set
     var creationTimestamp: Long = creationTimestamp
         private set
-    var expirationTimestamp: Long = expirationTimestamp
+    final override var expirationTimestamp: Long = expirationTimestamp
```
[the remark this answers is not in the document]
Comment (galganif) [reply java/arcs/core/entity/EntityBase.kt:44]: I didn't expose creationTimestamp in entity.kt, as it is not read anywhere from now.

File: java/arcs/core/entity/StorageAdapter.kt (hunk 1 above)
Comment (galganif) [reply java/arcs/core/entity/StorageAdapter.kt:75]: Adding a todo, but as you can see from here TTL is now wired in reference handle, and the timestamps are missing from core/entity/reference

**Pull request merged**